aten/src/ATen/BatchedTensorImpl.cpp

```diff
@@ -19,13 +19,18 @@ BatchedTensorImpl::BatchedTensorImpl(Tensor value, BatchDims bdims)
 
   const auto public_dims = value_.dim() - bdims_.size();
   const auto value_sizes = value_.sizes();
+  const auto value_strides = value_.strides();
   sizes_.clear();
   sizes_.reserve(public_dims);
+  strides_.clear();
+  strides_.reserve(public_dims);
   for (int64_t dim = 0; dim < public_dims; dim++) {
     auto actual_dim = actualDim(dim, /*wrap_dim=*/false);
     sizes_.push_back(value_sizes.at(actual_dim));
+    strides_.push_back(value_strides.at(actual_dim));
   }
   refresh_numel();
+  refresh_contiguous();
```

Comment:
Does this function need extra care?
What if I do `(B1, B0, 2, 3)` and permute to `(B0, B1, 2, 3)` it should still be contiguous even though the physical Tensor is not?

Reply:
refresh_contiguous() uses the `sizes_` and `strides_` field of the TensorImpl to determine if things are contiguous. Because the sizes_ and strides_ fields of BatchedTensorImpl are the per-example sizes and strides, the "is contiguous" computation is independent of the batch dims. 

> What if I do (B1, B0, 2, 3) and permute to (B0, B1, 2, 3) it should still be contiguous even though the physical Tensor is not?

Yes, it should still be contiguous. In this example, `sizes_ = [2, 3]` and `strides_ = [3, 1]`. `refresh_contiguous()` operates on those and those check out to be contiguous. I'll add an extra test case for this.

Reply:
For completeness:
- code for [refresh_contiguous](https://github.com/pytorch/pytorch/blob/f2eac5df1840b65b24ef2f1c3bd6332f654935b6/c10/core/TensorImpl.h#L1538-L1539) (we only really care about the first line that calls compute_contiguous)
- code for [compute_contiguous](https://github.com/pytorch/pytorch/blob/master/c10/core/TensorImpl.cpp#L94-L110)

Reply:
Sounds good! Thanks for the links!